plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.ts

```diff
@@ -42,15 +42,15 @@ export class CatalogIdentityClient {
    * Throws a NotFoundError or ConflictError if 0 or multiple users are found.
    */
   async findUser(query: UserQuery): Promise<UserEntity> {
-    const params = new URLSearchParams();
-    params.append('kind', 'User');
-
+    const conditions = ['kind=user'];
     for (const [key, value] of Object.entries(query.annotations)) {
-      params.append(`metadata.annotations.${key}`, value);
+      conditions.push(`metadata.annotations.${key}=${value}`);
```

Comment:
`URLSearchParams` handles encoding, need to url encode both the key and the value now.

Reply:
God call, had it on my mind but forgot